include/circt/Dialect/HW/HWOpInterfaces.td

```diff
@@ -18,13 +18,29 @@ include "mlir/IR/OpBase.td"
 
 def PortList : OpInterface<"PortList", []> {
   let cppNamespace = "circt::hw";
-  let description = "Operations which produce a unified port list representation";
+  let description = [{
+    Operations which produce a unified port list representation.
+    Users should override the port-indexed accessors.  No summary accessor is
+    provided by this interface as it is not assume that all ports have their
+    various constituent parts consecutive.  getPortList is the only summary 
+    accessor.  All per-port accessors are defaulted in terms of this, providing
```

Comment:
What "per-port accessors"? Need to clarify that you're (presumably) referring to ones in HWModuleLike. Or did you mean to move all of those accessors into this interface (as would make sense to me).